### ISSUE TITLE
Assign user permissions at a work level

### ISSUE THREAD

**Issue by smithjp** [samvera/hyrax#2302]
### Descriptive summary

When trying to assign reviewing permissions for a work to a specific user or group in hyrax 2.x as described in the documentation [(http://samvera.github.io/workflow_and_mediated_deposit.html#how-do-i-assign-reviewers-for-a-deposited-work-eg-etds-without-creating-lots-of-adminsets)](http://samvera.github.io/workflow_and_mediated_deposit.html#how-do-i-assign-reviewers-for-a-deposited-work-eg-etds-without-creating-lots-of-adminsets), a user with an approving workflow role for a specific work cannot see or review the work.  It appears that the entity_specific_responsibilities information is not being used to determine work permissions.

### Rationale

We would like to be able to have multiple departments use the same admin set instead of having duplicate admin sets for each department.  There would have to be a way for groups or individuals to be assigned to review subsets of works in the admin set based on department.

### Expected behavior

A user or group can be given permission to review a subset of works in an admin set.

### Actual behavior

A user or group can be given permission to review all works in an admin set.

**Comment by jcoyne** [comment]
@jeremyf is there a bug in the documentation here?  I'm not sure anyone has ever enabled Hyrax to work with EntitySpecificResponsibilities.

**Comment by jeremyf** [comment]
The Sipity query should allow this to work -- we assign permission to the
depositor of the work.

Jeremy Friesen

**Comment by smithjp** [comment]
From what I am seeing in the ability and Hyrax::Workflow::StatusListService classes, it looks like the entity specific responsibilities are not enabled in hyrax right now.  Are there any plans to do so?

**Comment by mjgiarlo** [comment]
@smithjp Not that I'm aware of, though perhaps the working group that @jeremyf and @laritakr have talked about spinning up re: roles/responsibilities may touch on this.

**Comment by jonathandixon** [comment]
I have implemented something very similar to what is described in this issue in our Hyrax 2.0 app. I don't recall all of the details, but I do see that I have a service that creates `Sipity::EntitySpecificResponsibility` entries. I will have to dig through our app and figure out if there are is any additional code required to get `Sipity::EntitySpecificResponsibility` to work (as of right now, I'm not finding any).

**Comment by smithjp** [comment]
@jonathandixon I have attempted to implement this by overriding the can_review_submissions method in the ability class and overriding some private methods in the status_list_service class.  The changes I made are here: [https://github.com/UNC-Libraries/hy-c/pull/4/files](https://github.com/UNC-Libraries/hy-c/pull/4/files).  Does that sound like what you did?

**Comment by jonathandixon** [comment]
I don't think you have to do any of that. All I did was create the `Sipity::EntitySpecificResponsibility` entries and Hyrax handled the rest. I didn't see this in the changeset you linked to. Here is a gist of what I did: https://gist.github.com/jonathandixon/232961b49fe14598d93a76de5f791caf.
